### PR TITLE
Litter model theory and implementation docs

### LINKED ISSUE #513: Litter model documentation
Once #509, #510, #511, and #512 are resolved the litter model will have pretty much reached minimum viable product stage. This would be an obvious time to properly document the model as well as the assumptions underlying it.

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

This PR adds documentation for the litter model documentation of theory. Let me know a) if what I've added makes sense and b) if there's anything else that would be good to add.

The updated documentation can be viewed here https://virtual-ecosystem.readthedocs.io/en/513-litter-model-documentation/

@TaranRallings as part of this I added some stuff to the animal model docs, let me know if you are happy with that. I'm also completely have for you to rework it as you see fit when you write the animal model documentation.

Fixes #513 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [ ] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [ ] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works
- [x] Relevant documentation reviewed and updated

### CODE REVIEW

**Review by davidorme — DISMISSED** [review]
I think the content is fine but its rather dense - big paragraphs of text - and a bit repetitive. 

I'm not sure what's helpful with doc PRs and this kind of feedback, but I think people should be prepared to submit an edited version? Would that be useful?

**Comment by jacobcook1995** [comment]
@davidorme yep I think that would be useful! It's not an easy thing to comment on using Github's (understandably) code focused tools

**Comment by davidorme** [comment]
@jacobcook1995 Partial edits. I've tried to do a couple of things:

* Make more use of bullet points and lists to help make the text more structured.
* I've moved all of the statements about future directions out of main text - where it tended to obscure what we are doing now - and into standalone admonitions.
* Shorten the text a bit.

If those pushes seem ok, let me know and I'll tackle the last big file on - whatever it was...

**Comment by jacobcook1995** [comment]
@davidorme think it's a real improvement thank you! Only thing is that the `:telescope:` doesn't seem to change the admonition when the docs build and instead just becomes an extra part of the title

**Review by vgro — CHANGES_REQUESTED** [review]
I really like how you describe the model and the references you provide for context. Your writing is very clear and easy to follow in most places.
I think the two models could be linked a bit better, it's not quite clear to me what happens where any why; for example why is there no environmental impact on the soil processes describes? Would it be worth starting with the litter model?

Here a few more specific comments:
Theory of biomass decay:
- I am not sure about 'biomass decay' as the main heading, in other places you refer to 'soil' or 'soil microbes' I think this might be a bit confusing
- I think it would be worth dropping above and belowground somewhere
- in the third paragraph, the biggest uncertainty IS / the biggest uncertainties ARE . Also maybe better to say 'in the litter model', just to make clear which one you are talking about. And a small typo in the last sentence: the model PROVIDES
- maybe worth mentioning the significantly less data for below ground in the last paragraph

Theory of the soil:
- maybe add to first sentence a short overview of the content and link to headings
- first paragraph: there seems to be a word missing: In response to this there has been a movement towards using soil carbon pool _definitions that defined by_ measurable physical and chemical properties
- could you say something about the relative contribution of the different carbon pools and the turnover rates? just roughly, is it daily or millennia?

Phosphorous pools
- you make reference to rain forests in the first sentence, can you say something about other systems, too?

the section on microbial representation could be a bit more specific, i.e. mention functional groups or enzymes (or why you decided not to use it)

I think it would be useful to mention somewhere how these nutrients are moved from one pool to the next (going back to residence times and rates) and to other part of the model

Theory of litter decay
 - above ground litter: all losses ARE assumed to be from...

**Comment by davidorme** [comment]
> Only thing is that the `:telescope:` doesn't seem to change the admonition when the docs build and instead just becomes an extra part of the title

I saw that and am sad about it... I wanted this:  🔭

**Comment by jacobcook1995** [comment]
@vgro I've made most of the changes you suggested. I haven't changed the description of the nitrogen and phosphorus pools, this is mainly because I am currently working on them and haven't settled on a final structure. I've made an issue (#606) to update this documentation later

File: docs/source/virtual_ecosystem/theory/soil/litter_theory.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+vs below-ground). These pools each have a characteristic decay rate, which gets modified
+by environmental conditions and for some of the pools by their lignin concentrations.
+Notably, these decay rates are not affected by the nitrogen and phosphorus
+concentrations of the pools. Instead, nitrogen and phosphorus concentrations effect the
+partitioning of input organic matter between litter pools, i.e. if the nutrient
+concentrations of a particular input are low a higher proportion of the input goes into
+slow decaying litter pools. This indirectly captures the impact of nitrogen and
+phosphorus chemistry on litter decay.
+
+The rest of this page gives provides details on the specific litter pools, the
```
Comment (vgro) [inline docs/source/virtual_ecosystem/theory/soil/litter_theory.md:40]: I think you missed adjusting this to the new layout

**Review by vgro — APPROVED** [review]
I like the changes that you made and I think it is in a good state as an intermediate overview. I think it's clear that this is not the final version from the intro.

**Comment by davidorme** [comment]
I think it's useful to have a common "in progress" admonition box for these pages? I mean, there's a chance they never go away, but then we can just have that at the top and keep the text cleaner. So:

```
In progress box

Text text text

Future directions box
```

**Comment by jacobcook1995** [comment]
@davidorme I've implemented your suggestions about admonitions, let me know what you think

**Review by TaranRallings — APPROVED** [review]
LGTM!

**Pull request merged**